pkg/providerid/providerid.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache v2.0 license.
+
+package providerid
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	ProviderName     string = "moc"
+	ProviderIDPrefix string = ProviderName + "://"
+)
+
+type HostType string
+
+const (
+	HostTypeVM        HostType = "vm"
+	HostTypeBareMetal HostType = "baremetal"
+)
+
+func FormatInstanceID(hostType HostType, machineName string) string {
```

Comment:
nit - This is the moc machine name and not the real machine name. I'm not sure if there's a way to make that clear through naming or maybe comments.